### PR TITLE
Don't handle params option in a special way in url_for helper

### PR DESCRIPTION
**Pull request by ilkkao**
### Summary

Potentially fixes https://github.com/rails/rails/issues/33248

### CODE REVIEW

**Comment by ilkkao** [comment]
@rafaelfranca there's a test that verifies that hash keys from `params`  will be merged back to options. 

Instead of removing this, I could just check the type

```
       if options.key? :params && options[:params].is_a?(Hash)
          params.merge! options[:params]
        end
```

That would also solve the problem I'm seeing.

File: actionpack/test/controller/base_test.rb [hunk 1]
```diff
@@ -193,10 +193,10 @@ def test_url_for_query_params_included
       action: "home",
       controller: "pages",
       only_path: true,
-      params: { "token" => "secret" }
+      parameter: "value"
```
Comment (rafaelfranca) [inline actionpack/test/controller/base_test.rb]: Could you keep `token: "secret"` here?

Comment (ilkkao) [reply actionpack/test/controller/base_test.rb]: Makes sense. Updated.

**Comment by ilkkao** [comment]
But then again, this functionality was originally added to support force SSL redirects. Removing it feels the best option as the current force SSL test pass.

**Comment by rafaelfranca** [comment]
Yes, it makes sense to me to remove. can you add a CHANGELOG entry?

**Comment by ilkkao** [comment]
@rafaelfranca changelog entry added

**Comment by ilkkao** [comment]
@rafaelfranca This would still be a useful fix for our team. Rebased the PR.

**Comment by oniofchaos** [comment]
Fix merge conflict on the changelog and 👍

**Pull request merged**